spec/lib/brexit_checker/results_audiences_spec.rb

```diff
@@ -2,81 +2,121 @@
 
 describe BrexitChecker::ResultsAudiences do
   describe "#populate_citizen_groups" do
-    let(:action1) { FactoryBot.build(:brexit_checker_action, :citizen, criteria: %w[living-uk living-row], grouping_criteria: %w[living-uk]) }
-    let(:action2) { FactoryBot.build(:brexit_checker_action, :citizen, criteria: %w[join-family-uk-yes], grouping_criteria: %w[living-uk]) }
-    let(:action3) { FactoryBot.build(:brexit_checker_action, :citizen, criteria: %w[nationality-uk], grouping_criteria: %w[living-uk]) }
-    let(:action4) { FactoryBot.build(:brexit_checker_action, :citizen, criteria: %w[visiting-driving], grouping_criteria: %w[visiting-eu]) }
-    let(:action5) { FactoryBot.build(:brexit_checker_action, :citizen, criteria: %w[studying-eu], grouping_criteria: %w[studying-eu]) }
-    let(:action6) { FactoryBot.build(:brexit_checker_action, :citizen, criteria: %w[living-uk visiting-ie], grouping_criteria: %w[living-uk visiting-ie]) }
-
-    let(:criteria1) { FactoryBot.build(:brexit_checker_criterion, key: "living-uk", text: "Living in the UK") }
-    let(:criteria2) { FactoryBot.build(:brexit_checker_criterion, key: "join-family-uk-yes", text: "You plan to join an EU or EEA family member in the UK") }
-    let(:criteria3) { FactoryBot.build(:brexit_checker_criterion, key: "nationality-uk", text: "You are a British national") }
-    let(:criteria4) { FactoryBot.build(:brexit_checker_criterion, key: "visiting-driving", text: "You need to drive abroad") }
-    let(:criteria5) { FactoryBot.build(:brexit_checker_criterion, key: "studying-eu", text: "You are studying in the EU") }
-    let(:criteria6) { FactoryBot.build(:brexit_checker_criterion, key: "living-row", text: "You do not live in the UK or the EU") }
-    let(:criteria7) { FactoryBot.build(:brexit_checker_criterion, key: "visiting-ie", text: "Visiting Ireland") }
-
-    let(:group_living_uk) { FactoryBot.build(:brexit_checker_group, key: "living-uk", heading: "Living in the UK") }
-    let(:group_visiting_eu) { FactoryBot.build(:brexit_checker_group, key: "visiting-eu", heading: "Visiting the EU") }
-    let(:group_studying_eu) { FactoryBot.build(:brexit_checker_group, key: "studying-eu", heading: "Studying in the EU") }
-    let(:group_visiting_ie) { FactoryBot.build(:brexit_checker_group, key: "visiting-ie", heading: "Visiting Ireland") }
-
-    let(:selected_criteria) { [criteria1, criteria2, criteria3, criteria4, criteria5] }
-    let(:actions) { [action1, action2, action3, action4, action5, action6] }
+    living_uk = FactoryBot.build(:brexit_checker_criterion, key: "living-uk")
```

Comment:
Was there a reason why these are no longer `let(:living_uk)` etc.?

Reply:
Every time factorybot builds the criteria or action via the let, it gets a new id. Because these models aren't active record objects, they are just hashes, if the value of the id changed the test failed. 

Because the criteria and actions are loaded from the yaml more than once, and passed around, even though the data is consistent the ID is changing. Which is maybe bad? Makes it hard to test at the very least. So in a later commit I refactored so that the criteria and actions are passed in at the initialise, so they remain unchanged - I reinstate the `let` blocks at that point.